.gitignore

```diff
@@ -3,3 +3,5 @@ src/proto/message.rs
 **/*.rs.bk
 Cargo.lock
 net-trace_*.txt
+TODOs
+.gitignore
```

Comment:
I don't think we want git to ignore the `.gitignore`?

Reply:
tbh, I committed this just to push my work so I could work from a different machine and forgot to remove it. In my personal projects, I actually like to have it ignored so I can ignore junk in my working directory without seeing `.gitignore` as one of the changed files. You can always `git add -f` if you actually need to commit a change to `.gitignore`, which is rare.

But I'll remove it. :)